examples/llava/phi4mm_convert_encoder_to_gguf.py

```diff
@@ -0,0 +1,318 @@
+import gguf
+import argparse
+import logging
+import sys
+import torch
+import json
+import os
+import numpy as np
+from typing import cast, ContextManager, Any, Iterator
+from pathlib import Path
+from torch import Tensor
+
+logger = logging.getLogger("phi4-mmproj")
+
+
+# https://huggingface.co/microsoft/Phi-4-multimodal-instruct/blob/main/modeling_phi4mm.py
+# https://huggingface.co/google/siglip-base-patch16-224/blob/main/preprocessor_config.json
+# https://github.com/EricLBuehler/mistral.rs/pull/1163/files
+SIGLIP_MODEL = {
+    "model_id": "google/siglip-base-patch16-224",
+    "image_size": 448,
+    "patch_size": 14, # I had very had time finding this number
+    "do_normalize": True,
+    "do_rescale": True,
+    "do_resize": True,
+    "image_mean": [
+        0.5,
+        0.5,
+        0.5
+    ],
+    "image_processor_type": "SiglipImageProcessor",
+    "image_std": [
+        0.5,
+        0.5,
+        0.5
+    ],
+    "processor_class": "SiglipProcessor",
+    "resample": 3,
+    "rescale_factor": 0.00392156862745098,
+    "size": {
+        "height": 224,
+        "width": 224
+    }
+}
+N_LAYERS = 27
+FEATURE_LAYER = -2
+HEAD_COUNT = 16
```

Comment:
Would be nice if this whole thing can be pulled from a JSON config instead of staying hard coded here, but the original model provides no such thing.